### DISCUSSION TITLE
Leaflet markers icons based on mode/band

### DISCUSSION THREAD

**Discussion by mbridak** [magicbug/Cloudlog#423] [Ideas / Feature Requests]
I'd like to see different markers on the map based on the mode. 

Most of the work can be done in leafembed.js
example code snippet below.

var icn = plotlist[i]["html"].split(" ")[8];
var plotll = new L.LatLng(plotlist[i].lat,plotlist[i].lng, true);
if (icn == 'CW'){
	var plotmark = new L.Marker(plotll, {icon: cwicon});
} else if (icn == 'SSB'){
	var plotmark = new L.Marker(plotll, {icon: ssbicon}); 
} else {
	var plotmark = new L.Marker(plotll);
}
![cloudlogmarkers](https://user-images.githubusercontent.com/49108421/75615220-f7da7d00-5af5-11ea-89bb-dc3a45b90543.png)

**Comment by magicbug** [comment]
Code theory isn't the problem the issue is time to generate multiple icons for each mode in the hobby, there's actually already an issue for this already #202 which goes with the idea of using different colours which I think might be the better way unless someone wants to create icons for the ADIF mode list?

**Comment by magicbug** [comment]
That said not ruling anything out, open to ideas and suggestions, if anyone has them!

**Comment by nr0q** [comment]
How about different icons for modes and color them based on band. I'm
putting my thinking cap on right now trying to imagine what icons would go
for each mode. I can picture what CW, FSK and PSK might look like.

CW have a dit and dah inside the pinhead, FSK could be a pair of
alternating vertical lines and PSK the hourglass looking shape for when the
phase flips in the signal. I"ll get to playing with my graphics program
here after I finish some coffee.

Matthew Chambers, CBT, NR0Q
Owner/Engineer
*M Chambers Communications Engineering LLC*
PO BOX 855, Moberly, MO 65270
Mobile (660)415-5620
www.mchambersradio.com

**Comment by nr0q** [comment]
Ok, so these aren't the prettiest icons I've ever made, but I'm thinking they wouldn't be up for interpretation and easy to read on a small screen. I made a sampling and attached them here. 
![AM-marker-icon-2x](https://user-images.githubusercontent.com/10331370/76475205-21d05100-63cc-11ea-9df9-9be0089ff094.png)
![FM-marker-icon-2x](https://user-images.githubusercontent.com/10331370/76475210-2563d800-63cc-11ea-90a1-02194af95038.png)
![SSB-marker-icon-2x](https://user-images.githubusercontent.com/10331370/76475214-28f75f00-63cc-11ea-8044-1d9d35e520c9.png)
![CW2-marker-icon-2x](https://user-images.githubusercontent.com/10331370/76475216-2ac12280-63cc-11ea-93b0-ba7ae7924c8a.png)
![PSK31-marker-icon-2x](https://user-images.githubusercontent.com/10331370/76475220-2dbc1300-63cc-11ea-993c-b1ba3b3a0217.png)
![JT65-marker-icon-2x](https://user-images.githubusercontent.com/10331370/76475226-314f9a00-63cc-11ea-945b-a8e805d6f14b.png)
![JT9-marker-icon-2x](https://user-images.githubusercontent.com/10331370/76475228-3280c700-63cc-11ea-96f2-d546a10a5d21.png)
![FT8-marker-icon-2x](https://user-images.githubusercontent.com/10331370/76475243-3b719880-63cc-11ea-8355-44bb5215397e.png)
![FT4-marker-icon-2x](https://user-images.githubusercontent.com/10331370/76475245-3e6c8900-63cc-11ea-9631-29e3aab9d734.png)

**Comment by nr0q** [comment]
I also want to note, I'm working on creating the various colored ones, but am wondering how many different colors would we want/need? 

- LF/MF (3 bands)
- HF (9 bands)
- VHF (3-4 bands)
- UHF (3 bands)

This would depend on how complex or simple we want to code this. If i were coding it, I'd go for simple at least to start with. have dark red for MF and a lighter red for 160m, then increasingly lighter shades of orange through yellow for the HF bands, a light green for 6m, and darker for 4m, and even dark for 2m, then light blue for 70cm, and get darker as we go up in freq with purple for 23cm and up? But that's just me. I'd like some input from others including the person that posted issue #202 .

**Comment by abrain** [comment]
> I also want to note, I'm working on creating the various colored ones

Creating a separate set of files per color is probably not necessary. If the icons are provided as SVG, they can be coloured through CSS. That also enables each instance to adapt the colors to their own needs.

But then there's still the question which colors to use initially. The following color scheme is used on https://pskreporter.info/, but I struggle to see a pattern behind it. Yours seems much more straight-forward.

![pskreporter-map-icons](https://user-images.githubusercontent.com/1263052/76517215-33156e00-645d-11ea-8e3a-e89b4ffb6bf1.png)

**Comment by mbridak** [comment]
I created a pull request with some code and icons. It's my first pull request so I THINK I did it right.

**Reply by acollins22** [reply]
Hi @mbridak,
Did you get anywhere with this? I for one would fine it very useful.


Thanks,

Andy.

**Reply by mbridak** [reply]
Yes I did. It worked rather well for me. I just created a new mode icon as I tried out a mode on the air and added it to my site. But then the main repo changed from teardrop icons to little pips, and I became rather fond of the pips, so just went with the 'stock' experience in the end. I found the teardrop icons to look like a big jumble mess if you had 'too many' contacts in a concentrated area.

**Reply by acollins22** [reply]
OK. Thanks for that. 

I really don't mind the pips but I would really like to colour them by the band so I can see at a glance where I've been getting to on what band.


Cheers,

Andy.
